### PR TITLE
chore(setup): make cmake fully permissive

### PR DESCRIPTION
**Pull request by sanchda**
Before I started mucking around with the cmake configuration, the `iast` `_native.so` extension was fully permissive--if it failed to build for whatever reason, then setup continued.

I had previously modified the logic to ensure that this extension was only circumstantially permissive (e.g., on macos).  At the time, my hope was to catch errors so that useful features could be more reliably included.  It turns out that this was incorrect.

I had missed an important point about optional extensions.  If an extension is truly optional, then failing the setup during any operation related to that extension is erroneous.

With these lessons in mind, I present a different way of structuring extension permissivity.

## Checklist

- [X] Change(s) are motivated and described in the PR description.
- [X] Testing strategy is described if automated tests are not included in the PR.
- [X] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [X] Change is maintainable (easy to change, telemetry, documentation).
- [X] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [X] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [X] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [ ] Title is accurate.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Testing strategy adequately addresses listed risk(s).
- [ ] Change is maintainable (easy to change, telemetry, documentation).
- [ ] Release note makes sense to a user of the library.
- [ ] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [ ] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -317,7 +317,16 @@ def try_strip_symbols(so_file):
...
+            try:
+                self.build_extension_cmake(ext)
+            except subprocess.CalledProcessError as e:
+                print("WARNING: Command '{}' returned non-zero exit status {}.".format(e.cmd, e.returncode))
+                if not ext.optional:
+                    raise
+            except Exception as e:
+                print("WARNING: An error occurred while building the CMake extension.")
+                if not ext.optional:
+                    raise
```
Comment (avara1986) [inline setup.py:329]: If I understand correctly, we can skip the build error, but then we call:

```python
self.try_strip_symbols(self.get_ext_fullpath(ext.name))
```

and if the file wasn't built correctly, we'll get a new error:
```
WARNING: Command '['cmake', '-S/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/ddtrace/appsec/_iast/_taint_tracking', '-B/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/cmake.linux-x86_64-cpython-39', '-DPYTHON_EXECUTABLE=/usr/local/bin/python3', '-DCMAKE_BUILD_TYPE=Release', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking', '-DLIB_INSTALL_DIR=/tmp/pip-install-7022oye3/ddtrace_4bd948db9d464ede9dd0b3ef09009660/build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking', '-DEXTENSION_NAME=_native.cpython-39-x86_64-linux-gnu.so']' returned non-zero exit status 1.
2511  strip: 'build/lib.linux-x86_64-cpython-39/ddtrace/appsec/_iast/_taint_tracking/_native.cpython-39-x86_64-linux-gnu.so': No such file <!----
```

Comment (sanchda) [reply setup.py:329]: Absolutely, good catch.

Comment (sanchda) [reply setup.py]: Made this more permissive, thank you for the report!

**Comment by sanchda** [comment]
This is superceded by #8203

**Pull request closed without merging**